### PR TITLE
storage: Export mount points etc to Anaconda

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

File: doc/anaconda.md [hunk 1]
```diff
@@ -102,3 +102,55 @@ case, Cockpit will use the type from "default_fsys_type".
...
+  "/dev/sda": {
+    "type": "filesystem",
+    "dir": "/",
+  },
+  "/dev/sdb": {
+    "type": "swap"
+  },
+  "/dev/sdc": {
+    "type": "crypto",
+    "child": {
```
Comment (mvollmer) [inline doc/anaconda.md:127]: Hmm, let's call this "content".

**Comment by mvollmer** [comment]
This doesn't have the passphrases yet. I want to do that separately.

File: pkg/storaged/anaconda.jsx [hunk 2]
```diff
@@ -19,10 +19,84 @@
...
+        for (const c of config) {
+            if (c[0] == "fstab") {
+                const d = client.strip_mount_point_prefix(decode_filename(c[1].dir.v));
+                if (d) {
+                    const sv = parse_subvol_from_options(decode_filename(c[1].opts.v));
+                    if (sv) {
+                        if (sv.pathname) {
+                            if (!subvols)
+                                subvols = { };
+                            subvols[sv.pathname] = d;
```
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:50]: These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19923-20240201-121849-281a7447-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx [hunk 3]
```diff
@@ -19,10 +19,84 @@
...
+                        }
+                    } else if (!dir) {
```
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:52]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19923-20240201-121849-281a7447-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx [hunk 4]
```diff
@@ -19,10 +19,84 @@
...
+                subvolumes: subvols
+            };
+    }
+
+    function block_info(block) {
+        if (block.IdUsage == "filesystem") {
+            return fstab_info(block.Configuration);
+        } else if (block.IdUsage == "swap") {
+            return {
+                type: "swap",
```
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:72]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19923-20240201-121849-281a7447-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx [hunk 5]
```diff
@@ -19,10 +19,84 @@
...
+            };
+        } else if (block.IdUsage == "crypto") {
```
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:74]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19923-20240201-121849-281a7447-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: test/verify/check-storage-anaconda [hunk 6]
```diff
@@ -198,6 +212,16 @@ class TestStorageAnaconda(storagelib.StorageCase):
...
         self.assertIn("noauto", m.execute("findmnt --fstab -n -o OPTIONS /sysroot/var"))
 
+        # Check exported mount point information.
+
+        self.expectExportedMountPoint({
+            "type": "filesystem",
+            "dir": "/var"
+        })
+
+        mpm = json.loads(b.call_js_func("window.localStorage.getItem", "cockpit_mount_points"))
```
Comment (mvollmer) [inline test/verify/check-storage-anaconda:222]: This code can be removed.

File: test/verify/check-storage-anaconda [hunk 7]
```diff
@@ -213,6 +237,89 @@ class TestStorageAnaconda(storagelib.StorageCase):
...
+                        "dir": "/"
+                    },
+                    "home": {
+                        "dir": "/home"
+                    },
+                }
+            }
+        })
+
+        testlib.sit()
```
Comment (mvollmer) [inline test/verify/check-storage-anaconda:321]: Debug

File: pkg/storaged/anaconda.jsx [hunk 8]
```diff
@@ -19,10 +19,85 @@
...
+                const d = client.strip_mount_point_prefix(decode_filename(c[1].dir.v));
+                if (d) {
+                    const sv = parse_subvol_from_options(decode_filename(c[1].opts.v));
+                    if (sv) {
+                        if (sv.pathname) {
+                            if (!subvols)
+                                subvols = { };
+                            subvols[sv.pathname] = { dir: d };
+                        }
+                    } else if (!dir) {
```
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:52]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19923-20240202-122001-15d323ee-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx [hunk 9]
```diff
@@ -19,10 +19,85 @@
...
+                subvolumes: subvols
+            };
+    }
+
+    function block_info(block) {
+        if (block.IdUsage == "filesystem") {
+            return fstab_info(block.Configuration);
+        } else if (block.IdUsage == "swap") {
+            return {
+                type: "swap",
```
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:72]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19923-20240202-122001-15d323ee-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx [hunk 10]
```diff
@@ -19,10 +19,85 @@
...
+            };
+        } else if (block.IdUsage == "crypto") {
```
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:74]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19923-20240202-122001-15d323ee-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by KKoukiou — COMMENTED** [review]
Tested it with plain subvolumes structure. Gives the expected result. Feel free to merge this, and I will test on monday for nested subvolumes usecases - since it's pretty 'normal' usecase.

File: pkg/storaged/anaconda.jsx (hunk 8 above)
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:52]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19923-20240202-135925-a1e0ec83-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx (hunk 9 above)
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:72]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19923-20240202-135925-a1e0ec83-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx (hunk 10 above)
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:74]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19923-20240202-135925-a1e0ec83-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx (hunk 8 above)
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:52]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19923-20240205-073014-a1e0ec83-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx (hunk 9 above)
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:72]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19923-20240205-073014-a1e0ec83-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx (hunk 10 above)
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:74]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19923-20240205-073014-a1e0ec83-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**